File: [0007][Easy][Reverse_Integer][math]/Reverse_Integer.py
# leetcode time     cost : 40 ms
# leetcode memory   cost : 13.7 MB
# Time  Complexity: O(logX)
# Space Complexity: O(1)
class Solution:
    def reverse(self, x: int) -> int:
        res = 0
        flag = 1 if x>0 else -1
        while x != 0:
            num = abs(x)%10
            res = res*10 + num
            x = abs(x)//10  
        return res*flag if - 2**31 < res <(2**31 -1 ) else 0
def main():
    inputX,expectRes = -3210, -123
    obj = Solution()
    
    result = obj.reverse(inputX)
    try:
        assert result == expectRes
        print("passed, result is follow expect:",result)
    except AssertionError as aError:
        print('failed, result >> ', result,"<< is wrong, ","expect is : ",expectRes)
    
if __name__ =='__main__':
    main() 

File: [0007][Easy][Reverse_Integer][math]/Reverse_Integer.cpp
// -*- coding: utf-8 -*- 
// leetcode time     cost : 0 ms
// leetcode memory   cost : 5.9 MB
// Time  Complexity: O(logX)
// Space Complexity: O(1)
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int reverse(int x) {
        int rev = 0;
        while (x != 0) {
            int pop = x % 10;
            x /= 10;
            if (rev > INT_MAX/10 || (rev == INT_MAX / 10 && pop > 7)) return 0;
            if (rev < INT_MIN/10 || (rev == INT_MIN / 10 && pop < -8)) return 0;
            rev = rev * 10 + pop;
        }
        return rev;
    }
};
/*
使用32位存储。
INT_MAX的值为+2147483647。
INT_MIN的值为-2147483648。
*/